lib/solargraph/complex_type.rb

```diff
@@ -241,7 +241,7 @@ def parse *strings, partial: false
               paren_stack += 1
             elsif char == ')'
               paren_stack -= 1
-              subtype_string += char if paren_stack == 0
```

Comment:
There wasn't much in the commit log - if you remember why this was there, happy to add more specs and ensure they pass as well.

Reply:
If I remember correctly, that condition was a quick fix for some wonky issue with the first iteration of type parameter support. It was seven years ago, so the specifics are lost to me. 🤷 Given the many examples of parametrized types in the specs, I'm plenty confident this won't introduce a regression.